pycheribuild/projects/run_qemu.py

```diff
@@ -222,18 +223,18 @@ def add_smb_dir(directory, target, share_name=None, readonly=False):
         if self._hasPCI:
             qemuCommand += ["-device", "virtio-rng-pci"]
 
-        if self.config.wait_for_debugger:
-            self.info("QEMU is waiting for GDB to attach (using `target remote :1234`)."
-                      " Once connected enter 'continue\\n' to continue booting")
-
+        if self.config.wait_for_debugger or self.config.debug_kernel:
+            gdb_port = randint(1200,1300) if self.config.gdb_random_port else 1234
```

Comment:
I think it's probably fine to always use a random port, so I'd add default=True to the gdb_random_port option.

For reliability (minimizing race conditions) you should use  `gdb_socket_placeholder = find_free_port()` which opens a socket with a random port, and assign `gdb_port = gdb_socket_placeholder.port`.
To make the port available to QEMU run `gdb_socket_placeholder.socket.close()` just before launching QEMU.

Reply:
I'm not sure I understood, this. can you fix it please?
Also, would be nice to automatically close the opened pane when qemu stops.. where is it better to close it, codewise?

Reply:
Fixed  (: